### PR TITLE
Add layout code and link to demo app

### PR DESCRIPTION
**Pull request by colleenmcginnis**
This PR does a few things:

- Adds XML layout code to examples where applicable. 
    - This is added automatically when using `node scripts/add-examples.js <product>`. 
    - In that script, we determine the name of the related layout file [using regex looking for `setContentView(R.layout.<name of layout>)`](https://github.com/mapbox/android-docs/blob/link-to-demo-app/scripts/add-examples.js#L295), then pulling the content of that file from the submodule and displaying in an `XmlCodeBlock` component.
    - This component includes a link to the relevant GitHub page for the layout file.
- Adds a link to Java and Kotlin activities.
    - I created a new `ExampleCodeBlock` that is built on top of the existing `ToggleableCodeBlock` component. It adds an `Activity` title to the code block (to distinguish it from the XML layout code block) and a link to the relevant GitHub page.
- Renames a lot of video files to match patterns in the `add-examples` script.
    - I ended up deleting and regenerating all example files using the script because it was more efficient. I renamed video files so if we ever want to do something like this again, there will be less manual work to do.
    - There are some file names (and therefore URL paths) that don't fit the pattens in the `add-examples` script because they were created before the script existed. To avoid having to create a lot of redirects, I kept those file names the same, but I created [a list](https://github.com/mapbox/android-docs/blob/link-to-demo-app/src/data/exceptions.json) of all of these so we can programmatically manage these if we want to make changes in the future.

Here's what the result looks like: 

screencapture-docs-tilestream-net-android-maps-examples-add-a-hillshade-layer-2019-08-19-10_35_24

### Checklist

* [x] QA all pages
    * [x] maps
    * [x] plugins
    * [x] java
    * [x] navigation
    * [x] vision (no links to GitHub because they don't exist in a demo app)
* [x] Get tests to pass
* [x] Investigate deleted md files
* [x] Clean up components
* [x] Clean up add-examples script
* [x] Run `npm test` locally to check for broken link warnings
* [x] Review by:
    * [x] @langsmith 
    * [x] @katydecorah 

cc @langsmith @zugaldia @mapbox/docs

### CODE REVIEW

**Comment by colleenmcginnis** [comment]
👋  @langsmith I'm going to work on these two items: 

>- [ ] Clean up components
>- [ ] Clean up add-examples script

but feel free to take a look at the [pages on staging](https://docs.tilestream.net/android/maps/examples/) any time. The layout and content should not change with the remaining clean up work I need to do. 

I went through and looked at every page to verify images/videos and GitHub links are working. Let me know if anything looks off to you!

**Comment by colleenmcginnis** [comment]
👋  @katydecorah can you take a look at these three files and let me know if you have any suggestions? (No rush. 🐌)

- [ ] [scripts/add-examples.js](https://github.com/mapbox/android-docs/pull/1041/files#diff-83a02aa4fc0f44f47294a9ee01237625)
- [ ] [src/components/example-code-block.js](https://github.com/mapbox/android-docs/pull/1041/files#diff-707c7650b1afae45d10ec2a3fecedc31)
- [ ] [src/components/xml-code-block.js](https://github.com/mapbox/android-docs/pull/1041/files#diff-7505e4551459ac01c91fcdb95e33a110)

File: src/components/toggleable-code-block.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { AppContext } from '../context.js';
```
Comment (katydecorah) [inline src/components/toggleable-code-block.js]: > import CodeSnippet from '@mapbox/react-code-snippet';

Let's update to `import CodeSnippet from '@mapbox/mr-ui/code-snippet';` and also remove `@mapbox/react-code-snippet` from package.json

**Comment by colleenmcginnis** [comment]
>How is example-code and https://github.com/mapbox/mapbox-android-demo kept in sync?
>Is it possible for code in /mapbox-android-demo to get ahead of example-code?

Yes. @langsmith opens a PR and manually bumps the submodule when there are significant changes or additions to the demo app. Once #1029 is up and running, I think we should consider automating this process -- opening a PR in this repo to bump the submodule when a change is merged into `/mapbox-android-demo`'s `master` branch. 

>For example, how can we make sure the fileName is pointing to the right file if updates are made to the /mapbox-android-demo repo? Should we get the latest commit from /mapbox-android-demo and add it to the fileName so that: https://github.com/mapbox/mapbox-android-demo/blob/master/MapboxAndroidDemo/src/main/res/layout/activity_basic_simple_mapview.xml becomes: https://github.com/mapbox/mapbox-android-demo/blob/b1252d4457555dde46ad81ccd4b4a9320f7b8ef8/MapboxAndroidDemo/src/main/res/layout/activity_basic_simple_mapview.xml to ensure the link doesn't break

How often are files renamed @langsmith? Would it be best to point to a specific commit or could we consider changing existing file names or file paths to be a "breaking change" in the demo app repo and require docs <--> android team coordination when that happens? 

>Double checking: where the added video files compressed?

No. I'm trying to limit the scope of this PR to get it out before the next time @langsmith wants to add additional examples. I will commit all compressed videos in a follow-up PR.

>I noticed that some thumbnail: images are hosted locally and others hosted on Imgur. We can acknowledge in a separate issue/PR, but I think it would be great to make sure this is consistent.

Yes let's look at this in a follow-up PR. This is happening because some examples were created before the `add-examples` script was written (those use an image IDs). The `add-examples` script looks up the image thumbnail being used in the demo app in [this file](https://github.com/mapbox/mapbox-android-demo/blob/master/MapboxAndroidDemo/src/main/res/values/urls_strings.xml) so those are on Imgur -- this keeps our site up to date with the demo app and makes it easier for developers running the demo app to cross reference.

**Comment by langsmith** [comment]
> How often are files renamed @langsmith?

Not often. File name is changed once every 4-5 months.

> Would it be best to point to a specific commit or could we consider changing existing file names or file paths to be a "breaking change" in the demo app repo and require docs <--> android team coordination when that happens?

Can we check that the file name exists when I bump the submodule and do the update sync betwen the demo app and this repo? If a file name has been changed, then a file no longer exists in this repo, and a flag is thrown.

**Comment by colleenmcginnis** [comment]
>Can we check that the file name exists when I bump the submodule and do the update sync betwen the demo app and this repo? If a file name has been changed, then a file no longer exists in this repo, and a flag is thrown.

So there are two things that could possibly be broken:
- The `View on GitHub` link that this PR adds
- The code in the code block

I think the problem @katydecorah is anticipating is that the file name changes in the demo app on the master branch and immediately breaks **the `View on GitHub` link** because that's pointing to the most recent commit master. 

The actual **code in the code block** won't break immediately because it's pointing to the version in the submodule. When you bump the submodule manually after a file name or path has changed, you will get an error and the build will not complete until you've updated it so I'm less worried about that piece.

File: src/components/toggleable-code-block.js (hunk 1 above)
Comment (colleenmcginnis) [reply src/components/toggleable-code-block.js]: Wow. We have a few of these that stuck around in this repo. https://github.com/mapbox/android-docs/issues/1048

**Comment by colleenmcginnis** [comment]
@katydecorah I opened some issues to address your concerns in follow-up PRs:

- https://github.com/mapbox/android-docs/issues/1050
- https://github.com/mapbox/android-docs/issues/1049
- https://github.com/mapbox/android-docs/issues/1048

wrt concerns about file name or file path changes -- based on the conversation above do you think we should refactor the approach here to include the commit or take the risk for now and revisit?

**Comment by katydecorah** [comment]
> based on the conversation above do you think we should refactor the approach here to include the commit or take the risk for now and revisit?

I think it would be ok to move forward. I wanted to flag it as a potential issue, but I think it makes sense to merge and assess as we go.

**Pull request merged**